Add prefix and suffix for tag

sawirricardo opened the pull request:
- [x] Changes have been thoroughly tested to not break existing functionality.
- [x] New functionality has been documented or existing documentation has been updated to reflect changes.
- [x] Visual changes are explained in the PR description using a screenshot/recording of before and after.

This will help folks who need to add prefix, but don't want to type it manually.

<img width="545" alt="image" src="https://github.com/filamentphp/filament/assets/37329575/3f5dc2d8-a7d6-4aa8-9fd6-62773effad8b">

sawirricardo committed:
add prefix and suffix for tag

danharrin commented:
This should probably be `tagPrefix()` and `tagSuffix()` rather than the other way round. Also for the properties and getter methods. Also, there shouldn't be any default `''` for the setter methods, its a required parameter

sawirricardo committed:
make suffix and prefix required

sawirricardo committed:
rename to be tagPrefix and tagSuffix

Feedback https://github.com/filamentphp/filament/pull/7461#issuecomment-1666436288

sawirricardo committed:
rename to tagPrefix and tagSuffix

danharrin commented on packages/forms/src/Components/TagsInput.php:
```diff
@@ -67,6 +71,20 @@ protected function setUp(): void
         $this->placeholder(__('filament-forms::components.tags_input.placeholder'));
     }
 
+    public function tagPrefix(string | Closure $prefix): static
```
```suggestion
    public function tagPrefix(string | Closure | null $prefix): static
```

danharrin commented on packages/forms/src/Components/TagsInput.php:
```diff
@@ -67,6 +71,20 @@ protected function setUp(): void
+        return $this;
+    }
+
+    public function tagSuffix(string | Closure $suffix): static
```
```suggestion
    public function tagSuffix(string | Closure | null $suffix): static
```

danharrin committed:
Apply suggestions from code review

danharrin commented:
Can you add some basic docs to `tags-input.md` please?

sawirricardo committed:
add docs about tagPrefix and tagSuffix

sawirricardo committed:
Merge branch 'add-prefix-suffix-to-tags-input' of https://github.com/sawirricardo/filament into add-prefix-suffix-to-tags-input

* 'add-prefix-suffix-to-tags-input' of https://github.com/sawirricardo/filament:
  Apply suggestions from code review

sawirricardo commented:
Updated the docs. Do you think the explanation is clear enough?

danharrin committed:
Update 14-tags-input.md

danharrin committed:
Update 14-tags-input.md

danharrin commented:
Thanks
